### PR TITLE
Add API Tests

### PR DESCRIPTION
**Pull request by mohdomama**
Relates to #438 
I am sending a fresh PR because codacy was not covering the previous one somehow :confused: 
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `rest-api` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:
- Use `requests` module to send post requests on the Rest API
- The post requests are send with different parameters each time and the response is checked with expected_responses.json file.
- I have tried to cover most of the functionality of the API in the tests. If anything is left, it can be easily added over time.

#### Travis Output
The following is the travis-ci output for my local fork :
https://travis-ci.org/mohdomama/badgeyay/jobs/329147958

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -34,7 +29,8 @@ before_script:
 
 script:
 - python app/main.py >> log.txt 2>&1  &
-# - nosetests app/tests/test.py -v --with-coverage | uncommented as tests are not ready yet.
+- sleep 2
```
Comment (abishekvashok) [inline .travis.yml]: Do we require this?

Comment (mohdomama) [reply .travis.yml]: Yes .. it takes some time for server to start. Hence `sleep 2`

**Comment by djmgit** [comment]
@mohdomama oh, you need to resolve conflicts in travis.yml file, I completely overlooked it :P

**Comment by mohdomama** [comment]
@S2606 @djmgit I have resolved the conflicts but the changes made in #450 are affecting the core functinality of the app. It should be resolved first. I have mentioned the same on gitter.

**Pull request merged**